packages/to-json-schema/src/convertSchema.ts

```diff
@@ -361,22 +368,28 @@ export function convertSchema(
       break;
     }
 
-    case 'enum': {
-      jsonSchema.enum = valibotSchema.options;
-      break;
-    }
-
+    case 'enum':
     case 'picklist': {
-      if (
-        valibotSchema.options.some(
-          (option) => typeof option !== 'number' && typeof option !== 'string'
-        )
-      ) {
-        handleError(
-          'An option of the "picklist" schema is not JSON compatible.',
-          config
-        );
+      let isStringArray = true;
```

Comment:
According to the JSON Schema specifications, if the enum contains mixed values (e.g., strings and numbers), the type property can be omitted to allow for this flexibility. However, if the enum values are all of the same type, it is considered good practice to include the type property for clarity and stricter validation.

Reply:
Thank you for the information and improvement. Can you link a reference where JSON schema mention the use of `.type` with `.enum`?

Reply:
I did read this somewhere but can't find it now, ig we can leave this for now. we can figure this out later

Reply:
I will have a look at [TypeBox](https://github.com/sinclairzx81/typebox) later to see how they handle enums.

Reply:
Seems like TypeBox is not using `.enum` at all. They use `.anyOf` instead:

![Screenshot-TypeBox-Enum](https://github.com/user-attachments/assets/e00c2878-f7c4-4a68-a37e-6fb5d1667f34)

Reply:
Looks like everyone has a different implementation, anyhow I will be writing some tests for my package soon. I will share the findings so we can take care of this later

Reply:
They changed it because of a config in AJV: https://github.com/sinclairzx81/typebox/pull/89

Every official example on the JSON schema website uses `.enum` without `.type`. So I would stick with the current implementation for now.